### ISSUE TITLE
A Couple Accessibility Issues with Talkback

### ISSUE THREAD

**Issue by stormdragon2976** [signalapp/Signal-Android#5533]
I have:
X [ ] searched open and closed issues for duplicates
- [ ] read https://github.com/WhisperSystems/Signal-Android/wiki/Submitting-useful-bug-reports

---
### Bug description

Describe here the issue that you are experiencing.
When receiving a call, Talkback does not read the in coming call information. Also, you kind have to guess where the slide to answer thing is and use 2 finger swipe to activate it. The standard phone app allows you to place a finger on the screen and focuses the answer bar then you can slide the 1 finger right to answer, Talkback will say swipe right to answer, up to send an sms, or left to decline.
Also, when ending a call, the phone's power button can not be used. It would be great if it were possible to use the power button to end a call, because it is difficult to find the end call button quickly, or if there is any near by noise, because talkback is reading through the phone's speaker instead of the external speaker.
If using the poser button is not possible, something lie iPhone's magic tap or whatever it is called would be ok too, where you just double tap the screen to end the call.
### Steps to reproduce
- using hyphens as bullet points
- list the steps
- that reproduce the bug

Activate talkback in settings, accessibility.
Try to answer a call using it.
**Actual result:** Describe here what happens after you run the steps above (i.e. the buggy behaviour)
Talkback does not read anything, as far as I can tell.
**Expected result:** Describe here what should happen after you run the steps above (i.e. what would be the correct behaviour)
Talkback should read the in coming caller info, and when you touch the screen it should say something like swipe right to answer.
### Device info

**Device:** Motorola Moto G first Generation
**Android version:** 4.4.2
**Signal version:** From playstore

**Comment by blindinsight** [comment]
Signal Accessibility Issues

Nexus5 running Android 6.0.1 Oct. 2016 security patch
TalkBack version 5.0.4.3, Usage Hints unchecked in TalkBack settings.

Main Screen
Using Explore By Touch navigation gestures swiping left to right to navigate to the next element on the screen, this is what I find starting top left corner:
"Signal" (App Title, nonfunctional)
"Search" (Not indicated as a button, but selectable with a double tap)
"More Options button" (Standard pullout menu)

"<Name of contact>, <time or day since last sent or received message>, <first 40 characters or so of the last message>, "Contact Photo Image Button" (this is selectable and takes you to the contact card for this person, however, couldn't this be more clearly identified within the conversation, and provide a setting to display contact photos.)
"New Conversation button" (clearly identified, however, if selected, I'm taken to a screen that doesn't indicate/separate which of my contacts has Signal. Could this be tabs like "All Contacts" and "Contacts with Signal”, or a way to filter in the “More options button” within this screen? Also each listed contact has their name and then "Contact photo button" If you really need to have that button there, could you label it "Go to contact button". Otherwise, examine how Google Messenger and Hangouts handle profile photos, and whether it's really necessary to be able to jump to the contact card in ALL these places.)

Conversation Screen:
Starting from top left corner swiping left to right
"Navigate Up button"
"<Name & phone#>" (not identified as a button, but selectable to modify settings for this contact/conversation. This is a duplicated way to get to these settings, which can also be accessed through the "More options button" that follows. In the conversation settings screen, there is an "unlabeled Button")
Signal Call" (not identified as a button, but launches call to this contact)

"More Options button", when expanded, there is:
"Disappearing messages, in list" (clearer direction is needed with this feature, and whether it's for the current message you're typing, if you select a duration before it disappears and press "OK", will the message be sent automatically, etc.)
"Add attachment"
"All Images"
"Conversation settings"
"Reset secure session"
"Mute notifications"

"Secure Message <Month, Date, & time>, in list"
"Contact Photo button" (same as main screen, but this appears on every message received from this contact, and makes it difficult to locate their message)
"<Month, Date, & Time> DeliveredSecure Message" (why on messages I receive, "Secure Message" is before the date/time, but on messages I send, date/time is spoken first, and "DeliveredSecure Message" is spoken after? FYI, "DeliveredSecure Message" is how it's displayed if I navigate by character)

**Request**
Instead of separating out the date/time, delivery status, & secure/unsecure status, could you include this info in the message?
For example, it would read something like this:
Received message: "Hey there. How are things? From <Contact's first name>, <date/time>, Secure Message"
Sent message: "Hi, doing well. From <My first name> <date/time> Secure Message Delivered"

I'm also a bit confused about the definition of "delivered". I've tested this with my fully sighted wife, and it seems to indicate via checkmarks when the receiving device receives and provides notification, but does NOT indicate that the message has been "read". This would be a nice indicator that once the conversation has been opened, that your sent messages change from "delivered" to "Read"? This feature could be toggled on/off in general and/or individual Contact settings within a conversation to provide the information of whether you've read the messages.
Continuing with the Conversation screen:
"toggle emoji keyboard button, out of list"
"Edit box, send Signal message"
If I write a message, the next left to right swipe would be "Signal" (not indicated as a button)
"Toggle attachment drawer button"
"Add attachment button"
If I double tap on "Add attachment button" this screen displays:
"Image" (plain text)
"Button unlabeled"
"Audio" (plain text)
"Button unlabeled"
"Video" (plain text)
"Button unlabeled"
"Contact" (plain text)
"Button unlabeled"
"Camera" (plain text)
"Button unlabeled"
"Location" (plain text)
"Button unlabeled"
"Space" (not selectable, and this isn't the word, but seems to indicate a couple additional spaces were added if I navigate by character)

Each of the "Button unlabeled" on this screen allow you to attach the item spoken before it. Why not combine the plain text and the "Button unlabeled" into single button?
 **Request**
Instead of only allowing me to attach a audio clip I've already saved, could you take the functionality from Google Messenger and allow me to record a message within the app, it be automatically attached to the message, allow me to preview or delete it before sending, and then I "Signal" when I'm ready.

**Comment by automated-signal** [comment]
GitHub Issue Cleanup:
See #7598 for more information.
